nifi-nar-bundles/nifi-atlas-bundle/nifi-atlas-reporting-task/src/test/java/org/apache/nifi/atlas/provenance/analyzer/TestGCSDirectory.java

```diff
@@ -0,0 +1,145 @@
+
+    @Test
+    public void testCompoundDirectory() {
+        String processorName = "PutHDFS";
+        String dirPath = "/dir1/dir2/dir3/dir4/dir5";
+        String expectedDirectoryQualifiedName = String.format("gs://%s%s/@%s", GCS_BUCKET, dirPath, ATLAS_NAMESPACE);
+
+        executeTest(processorName, dirPath, "dir5", "/dir1/dir2/dir3/dir4/", "dir4",
+                GCP_STORAGE_VIRTUAL_DIRECTORY, expectedDirectoryQualifiedName, AtlasPathExtractorUtil.GCS_VIRTUAL_DIR);
+    }
+
+    @Test
+    public void testRootDirectory() {
+        String processorName = "PutHDFS";
+        String dirPath = "/";
+        String expectedDirectoryQualifiedName = String.format("gs://%s@%s", GCS_BUCKET, ATLAS_NAMESPACE);
+
+        executeTest(processorName, dirPath, GCS_BUCKET, null, "/",
+                "gcp_storage_bucket", expectedDirectoryQualifiedName, AtlasPathExtractorUtil.GCS_BUCKET);
+    }
+
+    protected void executeTest(String processorName, String directory, String lastDirName, String parentPath, String parentName,
+                               String directoryType, String expectedDirectoryQualifiedName, String parentType) {
+        String transitUri = createTransitUri(directory);
+
+        ProvenanceEventRecord provenanceEvent = mockProvenanceEvent(processorName, transitUri);
+        AnalysisContext analysisContext = mockAnalysisContext();
+
+        NiFiProvenanceEventAnalyzer analyzer = NiFiProvenanceEventAnalyzerFactory.getAnalyzer(processorName, transitUri, PROVENANCE_EVENT_TYPE);
+        assertAnalyzer(analyzer);
+
+        DataSetRefs refs = analyzer.analyze(analysisContext, provenanceEvent);
+        assertAnalysisResult(refs, lastDirName, parentPath, parentName, directoryType, expectedDirectoryQualifiedName, parentType);
+    }
+
+    protected void assertAnalysisResult(DataSetRefs refs, String lastDirName, String parentPath, String parentName,
+                                        String directoryType, String expectedDirectoryQualifiedName, String parentType) {
+
+
+        Assertions.assertEquals(0, refs.getInputs().size());
+        Assertions.assertEquals(1, refs.getOutputs().size());
+
+        Referenceable directoryRef = refs.getOutputs().iterator().next();
+
+        Assertions.assertEquals(directoryType, directoryRef.getTypeName());
+        Assertions.assertEquals(expectedDirectoryQualifiedName, directoryRef.get(ATTR_QUALIFIED_NAME));
+        Assertions.assertEquals(lastDirName, directoryRef.get(ATTR_NAME));
+        Assertions.assertEquals(parentPath, directoryRef.get(AtlasPathExtractorUtil.ATTRIBUTE_OBJECT_PREFIX));
+
+        Referenceable bucketRef = (Referenceable) directoryRef.get(REL_PARENT);
+        if (parentPath != null) {
+            Assertions.assertNotNull(bucketRef);
+            Assertions.assertEquals(parentType, bucketRef.getTypeName());
+            Assertions.assertEquals(parentName, bucketRef.get(ATTR_NAME));
+        }
+    }
+
+    private String createTransitUri(String directory) {
```

Comment:
I think you can directly use the expectedDirectoryQualifiedName with a different name instead of using this in the executeTest method. Currently expectedDirectoryQualifiedName checks against something that is created in the test and not in the production code which makes the assertion useless and makes the test more difficult to understand.

Reply:
I think it should be different from the transitUri, and in case of testRootDirectory it won't work. I moved the creation of transitUris to the test cases. What do you think?